catalyst/core/callbacks/optimizer.py

```diff
@@ -70,14 +83,20 @@ def grad_step(
             optimizer_wds (List[float]): list of weight decay parameters
                 for each param group
             grad_clip_fn (Callable): function for gradient clipping
+            xla_args (Mapping[str, Any]): arguments to use in xla mode,
+                if not ``None`` then will be used xla optimization
+                function, default is ``None``
         """
         for group, wd in zip(optimizer.param_groups, optimizer_wds):
             if wd > 0:
                 for param in group["params"]:
                     param.data = param.data.add(-wd * group["lr"], param.data)
             if grad_clip_fn is not None:
                 grad_clip_fn(group["params"])
-        optimizer.step()
+        if xla_args is not None:
```

Comment:
let's define `callback._optimizer_step` and redefine it based on xla availability :)